File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        l, r = [1] * n, [1] * n
        for i in range(1, n):
            l[i] = l[i-1] * nums[i-1]
            r[-i-1] = r[-i] * nums[-i]
        return [l[i] * r[i] for i in range(n)] 

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        int res = 0;
        for (int num : nums) {
            if (!numSet.contains(num - 1)) {
                int start = num;
                int curr = 1;
                while(numSet.contains(num+curr)) curr++;
                res = max(curr, res);
            }
        }
        return res;
    }
};
